Full service regression tests

jroebu14 opened the pull request:
Resolves #NUMBER

**Overall change:**
Generates tests for all services based on current integration tests.

**Code changes:**

- _A bullet point list of key code changes that have been made._
- _When describing code changes, try to communicate **how** and **why** you implemented something a specific way, not just **what** has changed._

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added labels to this PR for the relevant pod(s) affected by these changes
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

Commit:
create scripts to generate integration tests for all services and page types

Commit:
add baseline regression tests

Commit:
add live radio summary tests

rhenshaw56 committed:
use GelPageGrid in story page

rhenshaw56 committed:
update snapshots

Commit:
live radio embed tests

simonsinclair committed:
Merge branch 'latest' into constrain-story-page

Commit:
add regression test GUI tool

Commit:
remove accidental changes

sareh committed:
Add fb:pages metadata tag to all pages

Commit:
integration test refactor with full service regression

Commit:
add all integration tests

Commit:
fix test:ci script

Commit:
refactor integration test set up. remove generated test files

Commit:
Merge branch 'refactor-integration-tests' into full-service-regression-tests

karinathomasbbc commented on src/integration/services.js:
```diff
@@ -0,0 +1,340 @@
+module.exports = {
```
I think we should use the Cypress config here - I got it working for the `generateRegressionTests` in the `regression-tests-across-all-services-and-pagetypes` branch
https://github.com/bbc/simorgh/pull/6148/files#diff-c9af4078b135bcf96e56bbf5d2d9d152

Commit:
add build script to generate full service regression tests

Commit:
reduce integration page type tests to one service

rhenshaw56 committed:
Merge branch 'latest' into constrain-story-page

sareh committed:
Merge branch 'latest' into add-meta-tag-fb

rhenshaw56 committed:
Merge branch 'latest' of https://github.com/bbc/simorgh into constrain-story-page

rhenshaw56 committed:
update integration test snapshots

rhenshaw56 committed:
Merge branch 'constrain-story-page' of https://github.com/bbc/simorgh into constrain-story-page

simonsinclair committed:
Merge branch 'latest' into constrain-story-page

sareh committed:
Integration tests for fb:pages

rhenshaw56 committed:
revised constraining logic for gel group 5

rhenshaw56 committed:
Merge branch 'latest' of https://github.com/bbc/simorgh into constrain-story-page

rhenshaw56 committed:
Merge branch 'constrain-story-page' of https://github.com/bbc/simorgh into constrain-story-page

rhenshaw56 committed:
updated snapshots

sareh committed:
Merge branch 'latest' into add-meta-tag-fb

FK78 committed:
Merge branch 'latest' into constrain-story-page

FK78 committed:
Merge branch 'latest' into constrain-story-page

EinsteinNjoroge committed:
update react helmet

DenisHdz committed:
Pass overtyped summary to story promo

DenisHdz committed:
Merge branch 'latest' of github.com:bbc/simorgh into overtyped-summaries

rhenshaw56 committed:
removed constrain logic in Grid

rhenshaw56 committed:
Merge branch 'latest' of https://github.com/bbc/simorgh into constrain-story-page

DenisHdz committed:
Fix typo

DenisHdz committed:
Merge branch 'latest' of github.com:bbc/simorgh into overtyped-summaries

DenisHdz committed:
Update snapshot

rhenshaw56 committed:
Merge branch 'constrain-story-page' of https://github.com/bbc/simorgh into constrain-story-page

DenisHdz committed:
Merge branch 'latest' into overtyped-summaries

simonsinclair committed:
Merge branch 'latest' into constrain-story-page

rebeccamcginn committed:
Refining data for live radio pages

DenisHdz committed:
Merge branch 'latest' of github.com:bbc/simorgh into overtyped-summaries

DenisHdz committed:
Update snapshots

DenisHdz committed:
Merge branch 'overtyped-summaries' of github.com:bbc/simorgh into overtyped-summaries

DenisHdz committed:
More snapshots

DenisHdz committed:
Snapshots

DenisHdz committed:
Merge branch 'latest' into overtyped-summaries

rhenshaw56 committed:
Merge branch 'latest' into constrain-story-page

EinsteinNjoroge committed:
Merge branch 'latest' into update-react-helmet

sareh committed:
Merge branch 'latest' into add-meta-tag-fb

rebeccamcginn committed:
Merge branch 'latest' of github.com:bbc/simorgh into onDemandATIWork

sareh committed:
Merge branch 'latest' into add-meta-tag-fb

EinsteinNjoroge committed:
Merge branch 'latest' into overtyped-summaries

FK78 committed:
Merge branch 'latest' into update-react-helmet

rebeccamcginn committed:
Updating integration snapshots for liveradio

rebeccamcginn committed:
Updating Chartbeat to handle new liveRadio pageData

rhenshaw56 committed:
modified for IE

rhenshaw56 committed:
restored env

rhenshaw56 committed:
updaated snapshots

rebeccamcginn committed:
Update ATI to match Radio data refactor

paruchurisilpa committed:
Merge branch 'latest' into add-meta-tag-fb

rhenshaw56 committed:
Merge branch 'latest' of https://github.com/bbc/simorgh into constrain-story-page

DenisHdz committed:
Add condition to check if the overtyped summary has whitespaces

EinsteinNjoroge committed:
react-helmet named import

rhenshaw56 committed:
Merge branch 'constrain-story-page' of https://github.com/bbc/simorgh into constrain-story-page

DenisHdz committed:
Merge branch 'overtyped-summaries' of github.com:bbc/simorgh into overtyped-summaries

DenisHdz committed:
Merge branch 'latest' of github.com:bbc/simorgh into overtyped-summaries

rebeccamcginn committed:
Merge branch 'latest' into onDemandATIWork

rhenshaw56 committed:
Merge branch 'latest' into constrain-story-page

DenisHdz committed:
Merge branch 'latest' into overtyped-summaries

rebeccamcginn committed:
Rename RadioPage to LiveRadioPage

DenisHdz committed:
Pass overtyped headline to index alsos

FK78 committed:
Merge branch 'latest' into constrain-story-page

DenisHdz committed:
Merge branch 'latest' of github.com:bbc/simorgh into overtyped-headlines-index-alsos

rebeccamcginn committed:
Refactoring propTypes for liveRadio

EinsteinNjoroge committed:
Merge branch 'latest' into overtyped-summaries

DenisHdz committed:
Merge branch 'latest' into overtyped-summaries

DenisHdz committed:
Merge branch 'latest' into overtyped-headlines-index-alsos

simonsinclair committed:
Merge branch 'latest' into constrain-story-page

FK78 committed:
Merge branch 'latest' into constrain-story-page

Commit:
Merge branch 'latest' into overtyped-headlines-index-alsos

Commit:
Merge branch 'latest' into overtyped-summaries

paruchurisilpa committed:
Merge branch 'latest' into constrain-story-page

sareh committed:
Merge branch 'latest' into constrain-story-page

thekp committed:
Merge branch 'latest' into overtyped-headlines-index-alsos

rebeccamcginn committed:
Removing hard-coded contentType from liveRadio ATI component

rebeccamcginn committed:
Merge branch 'latest' of github.com:bbc/simorgh into onDemandATIWork

karinathomasbbc committed:
Ensure integration tests run before pushing to branch

DenisHdz committed:
Merge branch 'latest' into overtyped-summaries

DenisHdz committed:
Update src/app/containers/StoryPromo/IndexAlsos/index.test.jsx

Co-Authored-By: Ruth Bochere <ruth.ogendi@andela.com>

EinsteinNjoroge committed:
Merge branch 'latest' into update-react-helmet

karinathomasbbc committed:
Retrigger builds

EinsteinNjoroge committed:
Bump psammead dependencies and update snapshots

amywalkerdev committed:
Merge branch 'latest' into add-meta-tag-fb

EinsteinNjoroge committed:
Remove duplicate packages

EinsteinNjoroge committed:
Merge pull request #6187 from bbc/update-react-helmet

update react helmet

karinathomasbbc committed:
Merge branch 'latest' into integration-tests-pre-push

amywalkerdev committed:
Merge branch 'latest' into add-meta-tag-fb

karinathomasbbc committed:
Merge pull request #6218 from bbc/integration-tests-pre-push

Ensure integration tests run before pushing to branch

DenisHdz committed:
Merge branch 'latest' into overtyped-headlines-index-alsos

simonsinclair committed:
Merge branch 'latest' into constrain-story-page

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into add-meta-tag-fb

merging

Commit:
updating integration tests

Commit:
Merge branch 'add-meta-tag-fb' of github.com:bbc/simorgh into add-meta-tag-fb

merging

rhenshaw56 committed:
Merge branch 'latest' of https://github.com/bbc/simorgh into constrain-story-page

rhenshaw56 committed:
Merge branch 'constrain-story-page' of https://github.com/bbc/simorgh into constrain-story-page

rebeccamcginn committed:
Merge branch 'latest' into onDemandATIWork

amywalkerdev committed:
Merge pull request #6170 from bbc/add-meta-tag-fb

Add missing fb:pages metadata tag to all pages

Commit:
position footer at bottom of page for OD and live radio

rebeccamcginn committed:
Merge branch 'latest' of github.com:bbc/simorgh into onDemandATIWork

PriyaKR committed:
Merge branch 'latest' into overtyped-summaries

PriyaKR committed:
Merge branch 'latest' into overtyped-headlines-index-alsos

paruchurisilpa committed:
Merge branch 'latest' into constrain-story-page

rhenshaw56 committed:
Merge branch 'constrain-story-page' of https://github.com/bbc/simorgh into constrain-story-page

RichardPK committed:
Matched cookie title within config to cypress' assertion

Commit:
Merge pull request #6221 from bbc/od-live-radio-position-footer-at-bottom

position footer at bottom of page for OD and live radio

PriyaKR committed:
Merge branch 'latest' into overtyped-headlines-index-alsos

RichardPK committed:
Merge branch 'latest' into fix-tigrinya-cookie-title-spacing

DenisHdz committed:
Merge branch 'latest' into overtyped-summaries

rebeccamcginn committed:
Merge branch 'latest' of github.com:bbc/simorgh into onDemandATIWork

rhenshaw56 committed:
Merge branch 'latest' into constrain-story-page

rebeccamcginn committed:
Updating integration snaps

jamesdonoh committed:
Merge pull request #6223 from bbc/fix-tigrinya-cookie-title-spacing

Changed Tigrinya cookie title within config/services to match cypress' assertion

DenisHdz committed:
Merge branch 'latest' into overtyped-headlines-index-alsos

DenisHdz committed:
Merge branch 'latest' into overtyped-summaries

rebeccamcginn committed:
Merge branch 'latest' into onDemandATIWork

simonsinclair committed:
Merge branch 'latest' into constrain-story-page

rebeccamcginn committed:
Merge pull request #6207 from bbc/onDemandATIWork

LiveRadio Refactor Regarding Ares Response

PriyaKR committed:
Merge branch 'latest' into overtyped-headlines-index-alsos

DenisHdz committed:
Merge branch 'latest' into overtyped-summaries

rhenshaw56 committed:
Merge branch 'latest' into constrain-story-page

DenisHdz committed:
Merge pull request #6213 from bbc/overtyped-headlines-index-alsos

Overtyped headlines for Index Alsos

DenisHdz committed:
Merge branch 'latest' into overtyped-summaries

karinathomasbbc committed:
Merge branch 'latest' of github.com:bbc/simorgh into full-service-regression-tests

amywalkerdev committed:
Merge pull request #6188 from bbc/overtyped-summaries

 Overtyped summaries in promos

amywalkerdev committed:
Merge branch 'latest' into constrain-story-page

Commit:
update integration tests

amywalkerdev committed:
Merge pull request #6152 from bbc/constrain-story-page

Constrain story page max-width

karinathomasbbc committed:
Delete generated test directory before running build script

karinathomasbbc committed:
Use correct registry for dependencies

karinathomasbbc committed:
Get services from Cypress config

karinathomasbbc committed:
Tidy up article tests
- removing service / pathname docs as it is not required
- deleting snapshots
- some article pages don't have images - make tests more robust

karinathomasbbc committed:
Tidy up front page tests
- removing service / pathname docs as it is not required
- deleting snapshots

karinathomasbbc committed:
Tidy up live radio tests
- removing service / pathname docs as it is not required
- deleting snapshots
- deleting tests which are no longer required

karinathomasbbc committed:
Update MAP tests to ensure consistency with other page type tests

karinathomasbbc committed:
Deleting service config as we're now using Cypress config

karinathomasbbc committed:
Adding common tests for MAPs

karinathomasbbc committed:
Update Story Page tests to ensure consistency with other page type tests

karinathomasbbc committed:
Improve the ATI tests - use snapshots to get the content of the ATI script

karinathomasbbc committed:
Rename test / build command

karinathomasbbc committed:
Reinstate article tests for individual page + update snapshots. Use correct test article path

karinathomasbbc committed:
Reinstate front page pathname

karinathomasbbc committed:
Reinstate korean live radio test + snapshots

karinathomasbbc committed:
Reinstate pidgin MAP test + update snapshots

karinathomasbbc committed:
Updating snapshots after renaming tests

karinathomasbbc committed:
Reinstate story page path name & associated snapshots

karinathomasbbc committed:
Remove test file since render.js has been deleted

karinathomasbbc committed:
Merge branch 'refactor-integration-tests' of github.com:bbc/simorgh into refactor-integration-tests

karinathomasbbc committed:
Updating snapshots

karinathomasbbc committed:
Deleting tests - they have been replaced by tests in /articles

karinathomasbbc committed:
Merge branch 'refactor-integration-tests' into full-service-regression-tests

karinathomasbbc committed:
Updates to common integration tests

karinathomasbbc committed:
Refactor article integration tests

karinathomasbbc committed:
Refactor front page integration tests

karinathomasbbc committed:
Refactor live radio integration tests

karinathomasbbc committed:
Refactor MAP integration tests

karinathomasbbc committed:
Refactor story integration tests

karinathomasbbc committed:
Remove no-cache from jest cli options

karinathomasbbc committed:
Reinstate AMP analytics tests

karinathomasbbc committed:
Update front page snapshots

karinathomasbbc committed:
Update live radio snapshots

karinathomasbbc committed:
Update MAP snapshots

karinathomasbbc committed:
Updating MAP snapshots

karinathomasbbc committed:
Update live radio snapshots

karinathomasbbc committed:
Updating snapshots

karinathomasbbc committed:
Reinstate common index

karinathomasbbc committed:
Updating snapshots

karinathomasbbc committed:
Updating snapshots

karinathomasbbc committed:
Update snapshots

karinathomasbbc committed:
Clean up snapshots

karinathomasbbc committed:
Adding a snapshot resolver for amp & canonical to control where tests are saved

karinathomasbbc committed:
Removing inline snapshots

karinathomasbbc committed:
Updating story test description

karinathomasbbc committed:
Updating snapshots

karinathomasbbc committed:
Use snapshots instead of inline snapshots

karinathomasbbc committed:
Removing inline snapshots

karinathomasbbc committed:
Merge branch 'refactor-integration-tests' into full-service-regression-tests

karinathomasbbc committed:
Update snapshots

karinathomasbbc committed:
Add line break

karinathomasbbc committed:
Removing snapshots

karinathomasbbc committed:
Reinstate path.join

karinathomasbbc committed:
Updating snapshots for unit tests

karinathomasbbc committed:
Merge branch 'refactor-integration-tests' into full-service-regression-tests

karinathomasbbc committed:
Update test - if there is no skip to content, or nav bar, then don't run tests

karinathomasbbc committed:
Fix failing test

karinathomasbbc committed:
Fix the import path for all generated tests

karinathomasbbc committed:
Remove --ci flag when running test:integration:all-services locally

karinathomasbbc committed:
Updating test names & corresponding snapshots

karinathomasbbc committed:
Updating test names & snapshots

karinathomasbbc committed:
Reinstate footer tests

karinathomasbbc committed:
Adding additional tests for footer links

karinathomasbbc committed:
Updating header tests

karinathomasbbc committed:
Update footer tests

karinathomasbbc committed:
Footer test for external link URL

karinathomasbbc committed:
Update dependency check

karinathomasbbc committed:
Adding missing dependencies

karinathomasbbc committed:
Adding the directory for autogenerated test files

karinathomasbbc committed:
Improving live radio & MAP media player tests
